app/cdap/components/PipelineDiff/DiffCanvas/PluginNode.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import { Handle, NodeProps, Position } from 'reactflow';
+import styled from 'styled-components';
+import classnames from 'classnames';
+import { IPipelineNodeData } from '../store/diffSlice';
+import { objectQuery } from 'services/helpers';
+import { pluginDiffColors } from '../util/helpers';
+import { DiffIcon } from '../DiffIcon';
+
+const NodeRoot = styled.div`
+  height: 110px;
+  width: 200px;
+  border: 3px solid ${(props) => pluginDiffColors[props.diffIndicator]};
+  border-radius: 5px;
+  background: white;
+  display: flex;
+  flex-direction: column;
+`;
+
+const HeaderRoot = styled.div`
+  width: 100%;
+  display: flex;
+`;
+
+const CustomIconRoot = styled.img`
+  &&& {
+    width: 25px;
+    height: 25px;
+    margin: 10px;
+  }
+`;
+
+const IconRoot = styled.div`
+  &&& {
+    font-size: 25px;
+    padding: 10px;
+  }
+`;
+
+const TitleContainer = styled.div`
+  flex-grow: 1;
+  padding-top: 10px;
+`;
+
+const TitleLabel = styled.label`
+  display: block;
+  color: ${(props) => pluginDiffColors[props.diffIndicator]};
+  font-size: 14px;
+  text-overflow: ellipsis;
+  overflow: hidden;
+`;
+
+const SubtitleLabel = styled(TitleLabel)`
+  font-size: 10px;
+  color: ${(props) => pluginDiffColors[props.diffIndicator] + '88'};
+`;
+
+const DiffIconContainer = styled.div`
+  display: flex;
+  flex-grow: 1;
+  justify-content: flex-end;
+  align-items: flex-end;
+  padding: 0 10px 10px 10px;
+`;
+
+const HandleRoot = styled(Handle)`
+  width: 14px;
+  height: 14px;
+  background-color: #4e5568;
+  border-radius: 100%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const AlertHandleRoot = styled(HandleRoot)`
+  left: 30px;
+`;
+
+const ErrorHandleRoot = styled(HandleRoot)`
+  left: 60px;
+`;
+
+const BottomHandleLabel = styled.span`
+  font-size: 11px;
+  position: absolute;
+  top: -16px;
+  color: ${(props) => pluginDiffColors[props.diffIndicator] + '40'};
+`;
+
+const PluginNode = ({
+  id,
+  data,
+  children,
+}: React.PropsWithChildren<NodeProps<IPipelineNodeData>>) => {
+  const diffIndicator = objectQuery(data, 'diffItem', 'diffIndicator') ?? 'default';
```

Comment:
Use optional chaining instead of object query https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
done

Reply:
I think you misunderstood. `objectQuery` is no longer necessary, do not use it.

Reply:
nevermind, somehow i was on the outdated pr... 🤦